clients/src/main/java/org/apache/kafka/common/config/internals/BrokerSecurityConfigs.java

```diff
@@ -89,4 +91,31 @@ public class BrokerSecurityConfigs {
             + "The broker will disconnect any such connection that is not re-authenticated within the session lifetime and that is then subsequently "
             + "used for any purpose other than re-authentication. Configuration names can optionally be prefixed with listener prefix and SASL "
             + "mechanism name in lower-case. For example, listener.name.sasl_ssl.oauthbearer.connections.max.reauth.ms=3600000";
+
+    public static class SaslEnabledMechanismsValidator implements ConfigDef.Validator {
```

Comment:
This is not a part of the public API (notice the `internals` in the package name) and Javadocs are overkill when the implementation is this straightforward IMO.

That said, it might be helpful to declare the class as `private` just to signal that it's not used anywhere outside of `BrokerSecurityConfigs`.

Reply:
thanks for your review